src/Middleware/Localization/src/RequestLocalizationMiddleware.cs

```diff
@@ -212,14 +213,23 @@ private static CultureInfo GetCultureInfo(StringSegment name, IList<CultureInfo>
 
             if (culture == null && fallbackToParentCultures && currentDepth < MaxCultureFallbackDepth)
             {
-                var lastIndexOfHyphen = cultureName.LastIndexOf('-');
+                if (AvailableCultures == null) {
+                    AvailableCultures = CultureInfo.GetCultures(CultureTypes.AllCultures);
```

Comment:
[Under the hood](https://github.com/dotnet/corefx/blob/d3911035f2ba3eb5c44310342cc1d654e42aa316/src/Common/src/CoreLib/System/Globalization/CultureInfo.cs#L611) this ends up [creating `CultureInfo` objects](https://github.com/dotnet/corefx/blob/4a7075f188b5777ccb519f2af9b8a284f4383357/src/Common/src/CoreLib/System/Globalization/CultureData.Windows.cs#L683), so instead of creating one expensive `CultureInfo` per request, we would create X on the first request, where X is however many cultures exist on the system. Since it's being cached it's possible that this is an acceptable trade-off, but I'll have to test it out and see how it affects the first request, ensure that it does in fact only affect the first request, and consult with some stake-holders to ensure they're all OK with this.

Before I go into all that, when you tested this locally was there a noticeable difference in time-to-first-request?

Reply:
@ryanbrandenburg I'll do the test asap.